src/enemy.asm

```diff
@@ -0,0 +1,442 @@
+;
+; Animation settings
+;
+.code
+squady_idle_animation:
+    .byte $04                               ; length frames
+    .byte $20                               ; starting tile ID
+    .byte $02                               ; attribute set
+    .byte $01                               ; padding x, z -> 1 tiles wide and high
+
+octi_idle_anim:
+    .byte $04                               ; length frames
+    .byte $04                               ; starting tile ID
+    .byte $03                               ; attribute set
+    .byte $02                               ; padding x, z -> 2 tiles wide and high
+
+
+; TODO: add different settings for different states
+squady:
+    .addr squady_idle_animation          ; HI Ivan!         ;  idle animation
+    ; .byte $00                  
+
+spacetopus:
+    .addr octi_idle_anim
+
+
+small_squad_army:
+    .addr squady                 ; type of enemy         PS: are you suffering already
+    .byte $40, $10              ; x position, y position  PPS: now? :*
+    .addr squady
+    .byte $c0, $10
+    .addr spacetopus
+    .byte $58, $32
+    .addr spacetopus
+    .byte $a8, $32
+    .addr squady
+    .byte $40, $50
+    .addr squady
+    .byte $c0, $50         
+
+;
+; Constants
+;
+NUM_ENEMIES = 6
+
+ANIMATION_SPEED = 8
+
+OFFSET_PADDING = 1
+OFFSET_TILE_ID = 2
+OFFSET_ATTRIBUTE_SET = 3       
+
+
+.segment "RAM"
+; 6 enemies, 5 bytes each
+current_enemy_set:  .res 40 ; 2 byte enemy address, 2 byte address, 1 byte anim state
```

Comment:
but 6x5 makes 30... :/

Reply:
I have an excellent explanation for that!